node/src/tasks/user_sends_file.rs

```diff
@@ -121,84 +239,26 @@ where
+
+        let file_key = file_metadata.file_key::<HashT<StorageProofsMerkleTrieLayout>>();
 
         // TODO: Check how we can improve this.
         // We could either make sure this scenario doesn't happen beforehand,
         // by implementing formatting checks for multiaddresses in the runtime,
         // or try to fetch new peer ids from the runtime at this point.
         if peer_ids.is_empty() {
-            info!(target: LOG_TARGET, "No peers were found to receive file {:?}", file_metadata.fingerprint);
+            info!(target: LOG_TARGET, "No peers were found to receive file key {:?}", file_key);
         }
 
-        // Iterates and tries to send file to peer.
-        // Breaks loop after first successful attempt,
-        // since all peer ids belong to the same BSP.
-        for peer_id in peer_ids {
-            for chunk_id in 0..chunk_count {
-                debug!(target: LOG_TARGET, "Trying to send chunk id {:?} of file {:?} to peer {:?}", chunk_id, file_key, peer_id);
-                let proof = match self
-                    .storage_hub_handler
-                    .file_storage
-                    .read()
-                    .await
-                    .generate_proof(&file_key, &vec![ChunkId::new(chunk_id)])
-                {
-                    Ok(proof) => proof,
-                    Err(e) => {
-                        return Err(anyhow::anyhow!(
-                            "Failed to generate proof for chunk id {:?} of file {:?}\n Error: {:?}",
-                            chunk_id,
-                            file_key,
-                            e
-                        ));
-                    }
-                };
-
-                let upload_response = self
-                    .storage_hub_handler
-                    .file_transfer
-                    .upload_request(peer_id, file_key.as_ref().into(), proof)
-                    .await;
-
-                match upload_response {
-                    Ok(_) => {
-                        debug!(target: LOG_TARGET, "Successfully uploaded chunk id {:?} of file {:?} to peer {:?}", chunk_id, file_metadata.fingerprint, peer_id);
-                    }
-                    Err(e) => {
-                        error!(target: LOG_TARGET, "Failed to upload chunk_id {:?} to peer {:?}\n Error: {:?}", chunk_id, peer_id, e);
-                        // In case of an error, we break the inner loop
-                        // and try to connect to the next peer id.
-                        break;
-                    }
-                }
-            }
-            info!(target: LOG_TARGET, "Successfully sent file {:?} to peer {:?}", file_metadata.fingerprint, peer_id);
-            return Ok(());
+        match self.send_chunks_to_provider(peer_ids, &file_metadata).await {
+            Some(result) => result,
+            None => Err(anyhow::anyhow!(
+                "Failed to send file key {:?} to any of the peers",
+                file_key
+            )),
```

Comment:
Why not just forwarding the error from `send_chunks_to_provider`? They should be more accurate.